oncotator/index/TabixIndexer.py

```diff
@@ -123,17 +124,41 @@ def index(destDir, inputFilename, fileColumnNumList=None, preset=None):
             return outputFilename
 
         outputFilename = os.path.join(destDir, string.join([fileName, ".tabix_indexed", fileExtension], ""))
-        # Copy the input file to output file.
-        shutil.copyfile(inputFilename, outputFilename)
 
         # Load the file into a tsvReader.
         if preset in ("gff", "bed", "sam", "vcf", "psltbl", "pileup"):
+            # Copy the input file to output file.
+            shutil.copyfile(inputFilename, outputFilename)
             tabix_index = pysam.tabix_index(filename=outputFilename, force=True, preset=preset)
         else:
-            # Have to specify min_size=0 in pysam 0.8.1 to get pysam to correctly output a .tbi file 
+
+            # Need to comment out the header line with a "#", so we cannot simply copy the file.
+            input_reader = GenericTsvReader(inputFilename)
+
+            output_writer = file(outputFilename, 'w')
+            output_writer.writelines(input_reader.getCommentsAsList())
+
+            # Add "#" for the header line.
+            output_writer.write("#")
+            field_names = input_reader.getFieldNames()
+            output_writer.write("\t".join(field_names))
+            output_writer.write("\n")
+            output_writer.flush()
+
+            # Write the rest of the file
+            # This might be too slow, since a raw reader would be pretty fast.
```

Comment:
I suspect this will in fact be very slow.  DictReaders are super slow in python from what I've seen.  Couldn't you just get a raw reader, as a byte to the front of it and the dump the rest unchanged? Or is it dealing with things like comma separated files here as well?  Indexing may not be a bottleneck anywhere though, so it might not be worth changing.